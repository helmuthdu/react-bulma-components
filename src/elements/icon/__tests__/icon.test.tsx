import { render } from '@testing-library/react';
import * as React from 'react';
import { Icon } from '..';

describe('Icon component', () => {
  it('should render with material icons', () => {
    const { container } = render(<Icon pack="mdi" icon="bars" size="large" />);
    expect(container.querySelector('.mdi')).toHaveClass('mdi', 'mdi-bars', 'mdi-48px');
  });
  it('should render with fontawesome', () => {
    const { container } = render(<Icon pack="fas" icon="bars" size="medium" />);
    expect(container.querySelector('.fas')).toHaveClass('fas', 'fa-bars', 'fa-2x');
  });

  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Icon icon="bars" className="other-class" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have different color', () => {
    const { container } = render(<Icon icon="bars" color="primary" />);
    expect(container.querySelector('.icon')).toHaveClass('has-text-primary');
  });

  it('should have a differentÂ size', () => {
    const { container } = render(<Icon icon="bars" size="large" />);
    expect(container.querySelector('.icon')).toHaveClass('is-large');
  });
});
