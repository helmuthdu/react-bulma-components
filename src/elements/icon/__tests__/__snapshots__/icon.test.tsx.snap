// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <span
    class="icon other-class is-normal"
  >
    <i
      class="mdi mdi-bars mdi-24px"
    />
  </span>
</DocumentFragment>
`;
